stablehlo/reference/ProcessGrid.cpp

```diff
@@ -182,6 +204,20 @@ void ProcessGrid::outfeed(ArrayRef<Tensor> inputs) {
   outfeed_.push(llvm::to_vector(inputs));
 }
 
+SmallVector<Tensor> ProcessGrid::recv(ChannelId channelId,
+                                      ProcessId processId) {
+  std::unique_lock<std::mutex> lock(sendRecvChannels_[channelId].mutex);
+  sendRecvReady_.insert(channelId);
+  sendRecvConditions_[channelId].notify_one();
+
+  if (!sendRecvConditions_[channelId].wait_for(
+          lock, std::chrono::seconds(3),
+          [&] { return !sendRecvChannels_[channelId].result.empty(); }))
+    llvm::report_fatal_error("recv timed out");
+
+  return sendRecvChannels_[channelId].result;
```

Comment:
Do we need to re-empty the result once we're done for subsequent recv's? Perhaps use move, or move this value to a temp var and clear it?

Reply:
Good call, updated.